### PR TITLE
Change to use asic_instance() API in all test cases

### PR DESCRIPTION
**Pull request by abdosi**
Initially we had two set of API doing same thing `get_asic()` and `asic_instance()`.
As part of https://github.com/Azure/sonic-mgmt/pull/3070/files#diff-ac7113ac4afb15af99807b123d079536d87e66389eab665362b294e2a760fa24L2382 
change was made `get_asic()` to return sonichost if asic Index is none. (this is needed if we want to run something on host for multi-asic platfroms). This change cause some of platform test to fail on single-asic platfroms since they  were calling API's present in Asic host but not in Sonic host. 

With this PR made following changes:

- Use `asic_instance()`  to always return  ASIC object

- Rename `get_asic()` to `get_asic_or_sonic_host()` which will return sonic host if asic_index is None else return asic host object. This API is useful if we want to execute/run something on host as well on asic in multi-asic platforms

- Updated all the testcases to use correct set of API's.

Fix the issue 
https://github.com/Azure/sonic-mgmt/issues/3103

How I verify:
On Single Asic no exception is found.

### CODE REVIEW

File: tests/common/devices/sonic.py [hunk 1]
```diff
@@ -1081,6 +1081,9 @@ def run_redis_cli_cmd(self, redis_cmd):
         cmd = "/usr/bin/redis-cli {}".format(redis_cmd)
         return self.command(cmd)
 
+    def get_docker_cmd(self, cmd, container_name):
```
[the remark this answers is not in the document]
Comment (abdosi) [reply tests/common/devices/sonic.py:1084]: Thanks updated.

**Review by neethajohn — APPROVED** [review]

**Review by judyjoseph — APPROVED** [review]

**Review by neethajohn — APPROVED** [review]

**Review by smaheshm — APPROVED** [review]
looks good

**Pull request merged**